### ISSUE TITLE
Failing to get OSD started

### ISSUE THREAD

**Issue by zcourts** [ceph/ceph-ansible#1561]
I've been trying to get a ceph cluster setup and failing to get any an OSD started.
I've tested collocating journal and bluestore scenarios.
With bluestore, there is never an error that I can find in any logs of `journalctl`.  With `journal_collocation` the service fails to start. 
Bash output:
```
root@bloch-kim-0005:~# journalctl -xe
--
-- Unit ceph-osd@0.service has failed.
--
-- The result is failed.
May 23 08:53:43 bloch-kim-0005 systemd[1]: ceph-osd@0.service: Unit entered failed state.
May 23 08:53:43 bloch-kim-0005 systemd[1]: ceph-osd@0.service: Failed with result 'exit-code'.
May 23 08:53:43 bloch-kim-0005 ceph-mgr[147617]: 2017-05-23 08:53:43.182763 7f333409c700 -1 mgr ms_dispatch mgrdigest v1
May 23 08:53:44 bloch-kim-0005 sshd[159335]: Failed password for root from 116.31.116.45 port 32262 ssh2
May 23 08:53:47 bloch-kim-0005 sshd[159335]: Failed password for root from 116.31.116.45 port 32262 ssh2
May 23 08:53:48 bloch-kim-0005 ceph-mgr[147617]: 2017-05-23 08:53:48.183043 7f333409c700 -1 mgr ms_dispatch mgrdigest v1
May 23 08:53:52 bloch-kim-0005 sshd[159335]: Failed password for root from 116.31.116.45 port 32262 ssh2
May 23 08:53:52 bloch-kim-0005 sshd[159335]: Received disconnect from 116.31.116.45 port 32262:11:  [preauth]
May 23 08:53:52 bloch-kim-0005 sshd[159335]: Disconnected from 116.31.116.45 port 32262 [preauth]
May 23 08:53:52 bloch-kim-0005 sshd[159335]: PAM 2 more authentication failures; logname= uid=0 euid=0 tty=ssh ruser= rhost=116.31.116.45  user=root
May 23 08:53:53 bloch-kim-0005 ceph-mgr[147617]: 2017-05-23 08:53:53.183273 7f333409c700 -1 mgr ms_dispatch mgrdigest v1
May 23 08:53:58 bloch-kim-0005 ceph-mgr[147617]: 2017-05-23 08:53:58.183554 7f333409c700 -1 mgr ms_dispatch mgrdigest v1
May 23 08:54:01 bloch-kim-0005 CRON[159346]: pam_unix(cron:session): session opened for user root by (uid=0)
May 23 08:54:01 bloch-kim-0005 CRON[159347]: (root) CMD (/usr/local/rtm/bin/rtm 55 > /dev/null 2> /dev/null)
May 23 08:54:01 bloch-kim-0005 CRON[159346]: pam_unix(cron:session): session closed for user root
May 23 08:54:03 bloch-kim-0005 ceph-mgr[147617]: 2017-05-23 08:54:03.183943 7f333409c700 -1 mgr ms_dispatch mgrdigest v1
May 23 08:54:03 bloch-kim-0005 systemd[1]: ceph-osd@0.service: Service hold-off time over, scheduling restart.
May 23 08:54:03 bloch-kim-0005 systemd[1]: Stopped Ceph object storage daemon osd.0.
-- Subject: Unit ceph-osd@0.service has finished shutting down
-- Defined-By: systemd
-- Support: http://lists.freedesktop.org/mailman/listinfo/systemd-devel
--
-- Unit ceph-osd@0.service has finished shutting down.
May 23 08:54:03 bloch-kim-0005 systemd[1]: Starting Ceph object storage daemon osd.0...
-- Subject: Unit ceph-osd@0.service has begun start-up
-- Defined-By: systemd
-- Support: http://lists.freedesktop.org/mailman/listinfo/systemd-devel
--
-- Unit ceph-osd@0.service has begun starting up.
May 23 08:54:03 bloch-kim-0005 ceph-osd-prestart.sh[159389]: OSD data directory /var/lib/ceph/osd/ceph-0 does not exist; bailing out.
May 23 08:54:03 bloch-kim-0005 systemd[1]: ceph-osd@0.service: Control process exited, code=exited status=1
May 23 08:54:03 bloch-kim-0005 systemd[1]: Failed to start Ceph object storage daemon osd.0.
-- Subject: Unit ceph-osd@0.service has failed
-- Defined-By: systemd
-- Support: http://lists.freedesktop.org/mailman/listinfo/systemd-devel
--
-- Unit ceph-osd@0.service has failed.
--
-- The result is failed.
May 23 08:54:03 bloch-kim-0005 systemd[1]: ceph-osd@0.service: Unit entered failed state.
May 23 08:54:03 bloch-kim-0005 systemd[1]: ceph-osd@0.service: Failed with result 'exit-code'.
May 23 08:54:08 bloch-kim-0005 ceph-mgr[147617]: 2017-05-23 08:54:08.184363 7f333409c700 -1 mgr ms_dispatch mgrdigest v1
May 23 08:54:13 bloch-kim-0005 ceph-mgr[147617]: 2017-05-23 08:54:13.184513 7f333409c700 -1 mgr ms_dispatch mgrdigest v1
root@bloch-kim-0005:~# ls -ahl /var/lib/ceph/osd/
total 8.0K
drwxr-xr-x  2 ceph ceph 4.0K Jan 19 14:59 .
drwxr-xr-x 11 ceph ceph 4.0K May 22 07:35 ..
```

The directory it complains about definitely doesn't exist.
All options in `osds.yml` are default except enabling one scenario with:
```
devices:
  - /dev/sda3
```
where `sda3` was formatted as xfs.

`all.yml`  is basically this:
```
---
upgrade_ceph_packages: true
ntp_service_enabled: true
ceph_test: false
ceph_origin: 'upstream' # or 'distro' or 'local'
use_installer: true 
ceph_stable: true # use ceph stable branch
cephx: true
max_open_files: 131072
# will go to monitor_interface if both are defined.
monitor_interface: '{{ ansible_default_ipv4.interface }}'
#monitor_address: 0.0.0.0
# set to either ipv4 or ipv6, whichever your network is using
ip_version: ipv4
mon_use_fqdn: false # if set to true, the MON name used will be the fqdn in the ceph.conf
net_mask: "{{ ansible_default_ipv4.network }}/{{ ansible_default_ipv4.netmask }}"
net_cidr: "{{ net_mask | ipaddr('net') }}"
public_network: "{{net_cidr}}"
cluster_network: "{{net_cidr}}"
```
Deploying to single ubuntu 16.04, haven't tried multiple servers yet.
All other ceph processes appear to have started ok.
```
root@bloch-kim-0005:~# ceph -s
    cluster c736f3db-7018-458d-8761-31d09f44b369
     health HEALTH_ERR
            88 pgs are stuck inactive for more than 300 seconds
            88 pgs stuck inactive
            88 pgs stuck unclean
            no osds
     monmap e2: 1 mons at {bloch-kim-0005=87.98.219.96:6789/0}
            election epoch 5, quorum 0 bloch-kim-0005
      fsmap e10: 1/1/1 up {0=bloch-kim-0005=up:creating}
        mgr active: bloch-kim-0005
     osdmap e9: 0 osds: 0 up, 0 in
            flags sortbitwise,require_jewel_osds,require_kraken_osds
      pgmap v10: 88 pgs, 4 pools, 0 bytes data, 0 objects
            0 kB used, 0 kB / 0 kB avail
                  88 creating
```
And
```
root@bloch-kim-0005:~# ceph osd tree
ID WEIGHT TYPE NAME    UP/DOWN REWEIGHT PRIMARY-AFFINITY
-1      0 root default
```

**Comment by guits** [comment]
can you link the playbook run log, please?

**Comment by zcourts** [comment]
See below:
[ceph-deploy-output.txt](https://github.com/ceph/ceph-ansible/files/1029871/ceph-deploy-output.txt)

**Comment by leseb** [comment]
@zcourts using a partition instead a raw device is a bit buggy, can you test with a raw device instead? e.g:  `/dev/sda`. Thanks.

**Comment by guits** [comment]
@zcourts any news with this issue ?

**Comment by zcourts** [comment]
No. I had one issue after another with `ceph-ansible`. I decided to try using `ceph-deploy` and then a custom ansible role to add customisations. Got stuck on some network issues (which I've yet to understand so can't explain). This was an experiment to see if we should switch from Gluster to make use of the S3 compatible API and as a result fell on the back burner as it's not a huge priority. I'll spend some more time this week looking at it so I'll try to do another deploy using the raw device instead. 

To be honest here, I was looking to configure Ceph in a similar way to the current Gluster setup where for Ceph OSDs were just given a directory to store it's data. We have a tiny cluster of 10/11 machines so was looking to share the resources rather than dedicate whole servers to Ceph.
I found that setup wasn't supported going from the PR https://github.com/ceph/ceph-ansible/pull/1472 - from the Ceph docs there was nothing I could find which indicated this wasn't a supported case. Suffice to say, I'm having a go at all the setup options I find to see which one is most suitable and Ceph taking a whole device on servers with one 1/2TB disk is not ideal currently.
